test/test.dart

```diff
@@ -194,4 +196,34 @@ Future<void> setupBaas() async {
   final client = await (cluster == null ? BaasClient.docker(baasUrl) : BaasClient.atlas(baasUrl, cluster, apiKey!, privateApiKey!, projectId!));
 
   baasApps.addAll(await client.getOrCreateApps());
-}
\ No newline at end of file
+}
+
+@isTest
+Future<void> syncTest(
+  String name,
+  FutureOr<void> Function(ApplicationConfiguration configuration) testFunction, {
+  String appName = 'flexible',
+  dynamic skip,
+}) async {
+  final url = Uri.tryParse(Platform.environment['BAAS_URL'] ?? 'https://realm-dev.mongodb.com');
+  final apiKey = Platform.environment['BAAS_API_KEY'];
+  final projectId = Platform.environment['BAAS_PROJECT_ID'];
+
+  if (skip == null) {
+    skip = url == null || apiKey == null || projectId == null;
+  }
+  else if (skip is bool) {
+    skip = skip || url == null || apiKey == null || projectId == null;
+  }
+  
+  test(name, () async {
+      final app = baasApps[appName] ?? baasApps.values.first;
+      final temporary = await Directory.systemTemp.createTemp('realm_test_');
+      final configuration = ApplicationConfiguration(
+        app.clientAppId,
+        baseUrl: url,
+        baseFilePath: temporary,
+      );
+      return await testFunction(configuration);
+  }, skip: skip);
```

Comment:
If baas is not setup, should we set `skip` to `"Baas is not setup"` or something similar to make it explicit why the test was skipped?

Reply:
not sure we should be overwritting that

Reply:
ok I will use a string skip in this case